src/projects/shared/api/circuit/src/internal/assembly/Prim.ts

```diff
@@ -70,14 +70,26 @@ export interface RectanglePrim {
     style: Style;
     ignoreHit?: boolean;
 }
+export interface TextPrim {
+    kind: "Text";
+
+    contents: string;
+    pos: Vector
+    angle: number
+    font: string;
+
+    style: Style;
+    ignoreHit?: true;
+}
 export type BaseShapePrim =
     | BezierCurvePrim
     | CirclePrim
     | CircleSectorPrim
     | LinePrim
     | PolygonPrim
     | QuadCurvePrim
-    | RectanglePrim;
+    | RectanglePrim
+    | TextPrim;
```

Comment:
I don't think TextPrim should be a "BaseShape", kinda semantically but also I think it doesn't need a "Style" attribute and that's what the "font" should cover (you can also add a attribute for color I guess)